File: Paladins.Api/Paladins.Api/Paladins.Service/Services/GeneralService.cs
using Paladins.Common.ClientModels.General;
using Paladins.Common.Interfaces.Clients;
using Paladins.Common.Interfaces.DataAccess;
using Paladins.Common.Interfaces.Repositories;
using Paladins.Common.Interfaces.Services;
using Paladins.Common.Models;
using Paladins.Common.Requests;
using Paladins.Common.Requests.Controllers;
using Paladins.Common.Responses;
using System.Collections.Generic;
using System.Threading.Tasks;

namespace Paladins.Service.Services
{
    public class GeneralService: IGeneralService
    {
        private readonly IGeneralClient _generalClient;
        private readonly IUnitOfWorkManager _unitOfWorkManager;

        public GeneralService(IGeneralClient generalClient, IUnitOfWorkManager unitOfWorkManager)
        {
            _generalClient = generalClient;
            _unitOfWorkManager = unitOfWorkManager;
        }

        public async Task<Response<List<GeneralChampionsSkinsClientModel>>> GetChampionSkinsAsync(ChampionSkinsRequest request)
        {
            var response = await _generalClient.GetClientChampionSkinsAsync(request);

            return new Response<List<GeneralChampionsSkinsClientModel>>() { Data = response };
        }

        public async Task<PagedResponse<ChampionModel>> GetChampionsAsync(PagedRequest request)
        {
            return await _unitOfWorkManager.ExecuteSingleAsync
                <IChampionRepository, PagedResponse<ChampionModel>>
                (u => u.GetChampionsAsync(request));
        }
        public async Task<PagedResponse<ItemModel>> GetItemsAsync(ItemPagedRequest request)
        {
            return await _unitOfWorkManager.ExecuteSingleAsync
                <IItemRepository, PagedResponse<ItemModel>>
                (u => u.getItemsAsync(request));
        }
    }
}

File: Mobile/app/src/main/java/com/fivegearszerochill/paladins/data/api/GeneralService.kt
package com.fivegearszerochill.paladins.data.api

import com.fivegearszerochill.paladins.domain.models.PagedResponseModel
import com.fivegearszerochill.paladins.domain.models.initial.ChampionModel
import com.fivegearszerochill.paladins.domain.models.initial.ItemModel
import com.fivegearszerochill.paladins.domain.requests.ItemPagedRequest
import com.fivegearszerochill.paladins.domain.requests.PagedRequest
import retrofit2.http.Body
import retrofit2.http.POST

interface GeneralService {
    @POST("/api/General/GetChampions")
    suspend fun getChampions(@Body request: PagedRequest): PagedResponseModel<ChampionModel>

    @POST("/api/General/GetItems")
    suspend fun getItems(@Body request: ItemPagedRequest): PagedResponseModel<ItemModel>
}
